common/app/common/dfp/PageskinAdAgent.scala

```diff
@@ -25,11 +25,25 @@ trait PageskinAdAgent {
       } else Seq.empty
     } else {
       val targetingMap = toMap(metaData.commercial.map(_.adTargeting(edition)).getOrElse(Set.empty))
-      val keywordTargeting = targetingMap.get("k") match {
-        case Some(values: MultipleValues) => values.values.toSeq
-        case _ => Seq.empty
-      }
-      candidates filter { sponsorship => sponsorship.keywords.intersect(keywordTargeting).nonEmpty }
+
+      // Little DRY helper
+      val targetingMapValues = ((map: Map[String, AdTargetParamValue], key: String) =>
+        map.get(key) match {
+          case Some(values: MultipleValues) => values.values.toSeq
+          case _ => Seq.empty
+        }
+      )
+
+      val keywordTargeting = targetingMapValues( targetingMap , "k" )
+      val serieTargeting = targetingMapValues( targetingMap , "se" )
+
+      candidates filter { sponsorship => {
+                           // Two ways of matching:
```

Comment:
Comments aren't needed. If you need to explain predicates more assign to well named variables.

Reply:
👌